cluster/engine.go

```diff
@@ -1080,7 +1072,7 @@ func (e *Engine) CreateContainer(config *ContainerConfig, name string, pullImage
 	e.CheckConnectionErr(err)
 	if err != nil {
 		// If the error is other than not found, abort immediately.
-		if (err != dockerclient.ErrImageNotFound && !engineapi.IsErrImageNotFound(err)) || !pullImage {
+		if (err != testErrImageNotFound && !engineapi.IsErrImageNotFound(err)) || !pullImage {
```

Comment:
In Line 75, there is an annotation that testErrImageNotFound is used for testing. While if that, I am afraid here we do not need this error.
Will `ContainerCreate` return an error `testErrImageNotFound`?
@nishanttotla